File: src/test/java/com/integreety/yatspec/e2e/teststate/TraceIdGeneratorShould.java
package com.integreety.yatspec.e2e.teststate;

import org.junit.jupiter.api.Test;

import static java.lang.Long.parseLong;
import static org.apache.commons.lang3.RandomUtils.nextInt;
import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

class TraceIdGeneratorShould {

    @Test
    public void generateTraceIdOfCorrectLength() {
        for (int count = 0; count < nextInt(1000, 2000); count ++) {
            assertThat(TraceIdGenerator.generate().length(), is(16));
        }
    }

    @Test
    public void generateHexadecimalValue() {
        for (int count = 0; count < nextInt(1000, 2000); count ++) {
            parseLong(TraceIdGenerator.generate(), 16);
        }
    }
}

File: src/integrationHttpTest/kotlin/io/lsdconsulting/lsd/distributed/interceptor/integration/data/TraceIdGeneratorShould.kt
package io.lsdconsulting.lsd.distributed.interceptor.integration.data

import io.lsdconsulting.lsd.distributed.interceptor.integration.data.TraceIdGenerator.generate
import org.apache.commons.lang3.RandomUtils
import org.hamcrest.CoreMatchers
import org.hamcrest.MatcherAssert
import org.junit.jupiter.api.Test

class TraceIdGeneratorShould {
    @Test
    fun generateTraceIdOfCorrectLength() {
        for (count in 0 until RandomUtils.nextInt(1000, 2000)) {
            MatcherAssert.assertThat(generate().length, CoreMatchers.`is`(16))
        }
    }

    @Test
    fun generateHexadecimalValue() {
        for (count in 0 until RandomUtils.nextInt(1000, 2000)) {
            generate().toLong(16)
        }
    }
}
